src/timessquare/services/page.py

```diff
@@ -64,18 +82,165 @@ async def render_page_template(
+        """Get the HTML for a page given a set of parameter values, first
+        from a cache or triggering a rendering if not available.
 
-        **Note:** ultimately this service method should rendering the
-        *computed* notebook, rather than merely the template-rendered notebook.
-        However, this functionality is a useful stop-gap for developing the
-        front-end.
+        Returns
+        -------
+        nbhtml : `NbHtmlModel` or `None`
+            The NbHtmlModel if available, or `None` if the executed notebook is
+            not presently available.
         """
         page = await self.get_page(name)
         resolved_parameters = page.resolve_and_validate_parameters(parameters)
-        rendered_notebook = page.render_parameters(resolved_parameters)
-        html = page.render_html(rendered_notebook)
-        return html
+
+        # First try to get HTML from redis cache
+        nbhtml = await self._html_store.get(
+            page_name=page.name, parameters=resolved_parameters
+        )
+        if nbhtml is not None:
+            self._logger.debug("Got HTML from cache")
+            return nbhtml
+
+        # Second, look if there's an existing job request. If the job is
+        # done this renders it into HTML; otherwise it triggers a noteburst
+        # request, but does not return any HTML for this request.
+        nbhtml = await self._get_html_from_noteburst_job(
+            page=page, resolved_parameters=resolved_parameters
+        )
+        if nbhtml is not None:
+            return nbhtml
+
+        return None
+
+    async def _get_html_from_noteburst_job(
+        self, *, page: PageModel, resolved_parameters: Mapping[str, Any]
+    ) -> Optional[NbHtmlModel]:
+        """Convert a noteburst job for a given page and parameters into
+        HTML (caching that HTML as well), and triggering a new noteburst
+        """
+        # Is there an existing job in the noteburst job store?
+        job = await self._job_store.get(
+            page_name=page.name, parameters=resolved_parameters
+        )
+        if not job:
+            self._logger.debug("No existing noteburst job available")
+            # A record of a noteburst job is not available. Send a request
+            # to noteburst.
+            await self._request_noteburst_execution(
+                page=page, resolved_parameters=resolved_parameters
+            )
+            return None
+
+        r = await self._http_client.get(
+            job.job_url, headers=self._noteburst_auth_header
+        )
+        if r.status_code == 200:
+            noteburst_response = NoteburstJobResponseModel.parse_obj(r.json())
+            self._logger.debug(
+                "Got noteburst job metadata",
+                status=str(noteburst_response.status),
+            )
+            if noteburst_response.status == NoteburstJobStatus.complete:
+                ipynb = noteburst_response.ipynb
+                assert ipynb
```

Comment:
`RuntimeError` here? (Arguably, this is a failure of the type system and a complete job should have a different type than `NoteburstJobResponseModel` by making it a child class, which will give you better type checking because then the type system will know the attributes a job will have if it's a completed job versus an in-progress job once you do a single branch on `isinstance`.)